src/lib/structures/symbols.ml

```diff
@@ -317,82 +296,209 @@ let name_to_string =
-  | Op BV2Nat -> "bv2nat"
-  | Op Tite -> "ite"
-  | Op Reach -> assert false
-  | True -> "true"
-  | False -> "false"
-  | Void -> "void"
-  | In (lb, rb) ->
-    Format.sprintf "%s , %s" (string_of_bound lb) (string_of_bound rb)
-
-  | MapsTo x ->  Format.sprintf "%s |->" (Var.to_string x)
-
-  | Lit lit -> string_of_lit lit
-  | Form form -> string_of_form form
-  | Let -> "let"
-
-  | Op (Optimize {order; is_max=true}) -> Format.sprintf "maximize(-,%d)" order
-  | Op (Optimize {order; is_max=false}) -> Format.sprintf "minimize(-,%d)" order
-
-let to_string_clean s = to_string ~show_vars:false s
-let to_string s = to_string ~show_vars:true s
-
-let print_clean fmt s = Format.fprintf fmt "%s" (to_string_clean s)
-let print fmt s = Format.fprintf fmt "%s" (to_string s)
+      Fmt.pf ppf "|%s|" s
+
+module AEPrinter = struct
+  let pp_operator ppf op =
+    match op with
+    (* Core theory *)
+    | Tite -> Fmt.pf ppf "ite"
+
+    (* Reals and Ints theories *)
+    | Plus -> Fmt.pf ppf "+"
+    | Minus -> Fmt.pf ppf "-"
+    | Mult -> Fmt.pf ppf "*"
+    | Div -> Fmt.pf ppf "/"
+    | Modulo -> Fmt.pf ppf "%%"
+    | Pow -> Fmt.pf ppf "**"
+    | Sqrt_real -> Fmt.pf ppf "sqrt_real"
+    | Sqrt_real_default -> Fmt.pf ppf "sqrt_real_default"
+    | Sqrt_real_excess -> Fmt.pf ppf "sqrt_real_excess"
+    | Int_floor -> Fmt.pf ppf "int_floor"
+    | Int_ceil -> Fmt.pf ppf "int_ceil"
+    | Max_real -> Fmt.pf ppf "max_real"
+    | Max_int -> Fmt.pf ppf "max_int"
+    | Min_real -> Fmt.pf ppf "min_real"
+    | Min_int -> Fmt.pf ppf "min_int"
+    | Integer_log2 -> Fmt.pf ppf "integer_log2"
+    | Integer_round -> Fmt.pf ppf "integer_round"
+
+    (* Reals_Ints theory *)
+    | Abs_int -> Fmt.pf ppf "abs_int"
+    | Abs_real -> Fmt.pf ppf "abs_real"
+    | Real_of_int -> Fmt.pf ppf "real_of_int"
+    | Real_is_int -> Fmt.pf ppf "real_is_int"
+
+    (* FixedSizedBitVectors theory *)
+    | Extract (i, j) -> Fmt.pf ppf "^{%d; %d}" i j
+    | Concat -> Fmt.pf ppf "@"
+    | BV2Nat -> Fmt.pf ppf "bv2nat"
+    | Int2BV n -> Fmt.pf ppf "int2bv[%d]" n
+    | BVnot -> Fmt.pf ppf "bvnot"
+    | BVand -> Fmt.pf ppf "bvand"
+    | BVor -> Fmt.pf ppf "bvor"
+
+    (* ArraysEx theory *)
+    | Get -> Fmt.pf ppf "get"
+    | Set -> Fmt.pf ppf "set"
+
+    (* DT theory *)
+    | Record -> Fmt.pf ppf "@Record"
+    | Access s -> Fmt.pf ppf "@Access_%a" Hstring.print s
+    | Constr s -> Hstring.print ppf s
+    | Destruct (s, g) ->
+      Fmt.pf ppf "%s%a" (if g then "" else "!") Hstring.print s
+
+    (* Float theory *)
+    | Float -> Fmt.pf ppf "float"
+    | Fixed -> Fmt.pf ppf "fixed"
+
+    | Not_theory_constant -> Fmt.pf ppf "not_theory_constant"
+    | Is_theory_constant -> Fmt.pf ppf "is_theory_constant"
+    | Linear_dependency -> Fmt.pf ppf "linear_dependency"
+
+    | Optimize {order; is_max=true} ->
+      Fmt.pf ppf "maximize(-,%d)" order
+    | Optimize {order; is_max=false} ->
+      Fmt.pf ppf "minimize(-,%d)" order
+    | Reach -> assert false
```

Comment:
I will admit I have zero clue what this operator is, do you know?

Reply:
Actually I checked. It's completely unused. Both legacy and Dolmen parser never produces such symbol. Same situation for the symbol `Fixed`. It was supposed to be used by the arithmetic theory in `arith.ml`:
```
    | Sy.Op Sy.Fixed, _ ->
      (* Fixed-Point arithmetic currently not implemented *)
      assert false
```

Reply:
Right well, we should probably remove these then.

Reply:
I plan to do later ;)